Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -2913,6 +2928,43 @@ bool RenderLayerBacking::isDirectlyCompositedImage() const
     return false;
 }
 
+bool RenderLayerBacking::isUnscaledBitmapOnly() const
+{
+    if (!is<RenderImage>(renderer()) && !is<RenderHTMLCanvas>(renderer()))
+        return false;
+
+    if (m_owningLayer.hasVisibleBoxDecorationsOrBackground())
+        return false;
+
+    auto contents = contentsBox();
+    if (contents.location() != LayoutPoint(0, 0))
+        return false;
+
+    if (is<RenderImage>(renderer())) {
+        // FIXME: Test image path.
```

Comment:
Please do! Need a test with an image that succeeds, and images that fail because they are SVG, or use object-fit or object-position to get scaled or translated.